.docs/user-guide/configuration.md
# Configuration

What are the Framework's configuration options?

---

## Overview

This page reviews how to configure the ScaleIO-Framework. We will first cover the
command line option and delving into the details of more advanced settings.

## Basic Command Line Options

Basic command line options available...

`-loglevel=[debug|info|warn|error]`  
Optional: Set the logging level for the application. Default: info

`-debug=[true|false]`  
Optional: Debug mode prevents the reboot so the logs dont get cycled. Default: false

`-rexray.branch=[stable|unstable|staged]`  
Optional: Which branch to grab the REX-Ray package from. Default: stable

`-rexray.version=[specific version|latest]`  
Optional: Which version to install from the provided branch. You can provide a
specific version such as 0.6.3. The default is the last build. Default: latest

`-isolator.binary=[specific version]`  
Optional: The URL for which Mesos Module DVDI to install. Default: latest supported

`-rest.address=`  
Optional: *Highly recommend not to modify* Mesos scheduler REST API address.
Default: Dynamically calculated.

`-rest.port=<port>`
Required: *Highly recommend using the dynamic Marathon port* Mesos scheduler
REST API port.

`-uri=<mesos master uri>`
Required: Mesos scheduler API URL. This is the Mesos Master HTTP API endpoint.

`-executor.cpu.factor=[float value]`  
Optional: Fudge factor for effective CPU available. This allows overhead/reserve.
A value of 1.0 means none of the CPU has been reserved. Default: 1.0

`-executor.mem.factor=[float value]`  
Optional: Fudge factor for effective memory available. This allows overhead/reserve.
A value of 1.0 means none of the memory has been reserved. Default: 1.0

`-store.type=[consul|etcd|zk|boltdb]`  
Optional: The type of keyvalue store to use. Default: zk

`-store.uri=[uri to connect to the key store based on store.type]`  
Optional: Store URI to connect with. When the value is an "empty string", the
framework will dynamically determine the zookeeper endpoints.
Default: "empty string"

`-scaleio.clustername=[cluster name]`  
Optional: ScaleIO Cluster Name. Default: scaleio

`-scaleio.clusterid=`  
Optional: ScaleIO Cluster ID. The priority of this flag supersedes scaleio.clustername
if used. Default: "empty string"

`-scaleio.lbgateway=[uri for the scaleio gateway]`  
Optional: Great to use if your ScaleIO gateway is clustered and load balanced.
If the value is "empty string", it is assumed that the ScaleIO is on the primary
MDM node. Default: "empty string"

`-scaleio.protectiondomain=[default pd name]`  
Optional: The default ScaleIO Protection Domain Name used. Default: default

`-scaleio.storagepool=[default sp name]`  
Optional: The default ScaleIO StoragePool Name used. Default: default

`-scaleio.password=[password]`  
Required: ScaleIO Admin Password. This is used to install all ScaleIO packages
that require a ScaleIO admin password, the password to access the ScaleIO gateway,
and etc. Default: Scaleio123

`-scaleio.preconfig.primary=<fqdn/ip of pri MDM node>`  
Required: FQDN or IP of the pre-configured Pri MDM Node. Requires Sec and TB MDM
nodes to be pre-configured

`-scaleio.preconfig.secondary=<fqdn/ip of sec MDM node>`  
Required: FQDN or IP of the pre-configured Sec MDM Node. Requires Pri and TB MDM
nodes to be pre-configured

`-scaleio.preconfig.tiebreaker=<fqdn/ip of sec MDM node>`  
Required: FQDN or IP of the pre-configured Tiebreaker MDM Node. Requires Pri and
Sec MDM nodes to be pre-configured

`-scaleio.preconfig.gateway=[fqdn/ip of ScaleIO gateway node]`  
Optional: Used to set a separate ScaleIO Gateway node. If the value is "empty string",
the Primary MDM node is assumed to have the Gateway installed on it.
Default: "empty string"

`-scaleio.ubuntu14.mdm=[URL for DEB]`  
Optional: ScaleIO MDM Package for Ubuntu 14.04. Default: package tested in the release

`-scaleio.ubuntu14.sds=[URL for DEB]`  
Optional: ScaleIO SDS Package for Ubuntu 14.04. Default: package tested in the release

`-scaleio.ubuntu14.sdc=[URL for DEB]`  
Optional: ScaleIO SDC Package for Ubuntu 14.04. Default: package tested in the release

`-scaleio.ubuntu14.lia=[URL for DEB]`  
Optional: ScaleIO LIA Package for Ubuntu 14.04. Default: package tested in the release

`-scaleio.ubuntu14.gw=[URL for DEB]`  
Optional: ScaleIO Gateway Package for Ubuntu 14.04. Default: package tested in the release

`-scaleio.centos7.mdm=[URL for RPM]`  
`-scaleio.rhel7.mdm=[URL for RPM]`
Optional: ScaleIO MDM Package for RHEL7/CentOS7. Currently RHEL7 and CentOS7 are the same.
Default: package tested in the release

`-scaleio.centos7.sds=[URL for RPM]`
`-scaleio.rhel7.sds=[URL for RPM]`
Optional: ScaleIO SDS Package for RHEL7/CentOS7. Currently RHEL7 and CentOS7 are the same.
Default: package tested in the release

`-scaleio.centos7.sdc=[URL for RPM]`
`-scaleio.rhel7.sdc=[URL for RPM]`
Optional: ScaleIO SDC Package for RHEL7/CentOS7. Currently RHEL7 and CentOS7 are the same.
Default: package tested in the release

`-scaleio.centos7.lia=[URL for RPM]`
`-scaleio.rhel7.lia=[URL for RPM]`
Optional: ScaleIO LIA Package for RHEL7/CentOS7. Currently RHEL7 and CentOS7 are the same.
Default: package tested in the release

`-scaleio.centos7.gw=[URL for RPM]`
`-scaleio.rhel7.gw=[URL for RPM]`
Optional: ScaleIO Gateway Package for RHEL7/CentOS7. Currently RHEL7 and CentOS7 are the same.
Default: package tested in the release

## Advanced Command Line Options

Not going to lie... some of these command line option descriptions will be left
intentional vague so you don't shoot yourself in the foot. No point in hiding them
because you can just run a help on the command line. So in the interest in full
disclosure, here they are.

Most of these options are highly recommended *NOT* to modify and/or use.

`-experimental=[true|false]`  
Optional: Sets the application to experimental mode. Default: false

`-executor.altpath=[http path]`  
Optional: This is to override the default location of the scaleio-executor binary.
Default: "empty string"

`-executor.cpu.mdm=[float value]`  
Optional: CPU resources consumed by ScaleIO MDM software.
These are either the primary, secondary, or tiebreaker MDM nodes. Modifying
these values may not earmark actually CPU consumed by the ScaleIO software. Default: 1.5

`-executor.cpu.non=[float value]`  
Optional: CPU resources consumed by ScaleIO Non-MDM software.
These are data nodes only. Modifying these values may not earmark actually CPU
consumed by the ScaleIO software. Default: 0.5

`-executor.mem.mdm=[int value]`  
Optional: Memory resources (MB) consumed by ScaleIO MDM software.
These are either the primary, secondary, or tiebreaker MDM nodes. Modifying
these values may not earmark actually memory consumed by the ScaleIO software.
Default: 3072

`-executor.mem.non=[int value]`  
Optional: Memory resources (MB) consumed by ScaleIO MDM software.
These are data nodes only. Modifying these values may not earmark actually memory
consumed by the ScaleIO software. Default: 512

`-user=[user to run the process under]`  
Optional: The User account the framework is running under. Root is required in
order to run administrative functions like installation of packages. Default: root

`-hostname=[hostname where the scheduler is running]`  
Optional: The Hostname where the framework runs. This is only used for setting up
a mock scheduler. Default: dynamically discovered

`-role=[role name]`  
Optional: *Currently not supported* Framework role to register with the Mesos master.
This will allow for multiple ScaleIO clusters to run in the same Mesos cluster. Default: scaleio

`-scaleio.apiversion=[ScaleIO API Version]`  
Optional: ScaleIO API Version. Matches the API version of the ScaleIO software
the framework installs. Default: 2.0

`-store.delete=<true|false>`  
Optional: Helper function that deletes ScaleIO Framework Key/Value Store. Default: false

`-store.dump=<true|false>`  
Optional: Helper function that dumps ScaleIO Framework Key/Value Store. Default: false

`-store.add.key=<key to add to store>`  
Optional: Modify a select store key. Default: "empty string"

`-store.add.value=<value to add to the store>`  
Optional: Set the values to the key provided by store.key. Default: "empty string"

`-store.del.key=<key to delete>`  
Optional: Delete a select store key. Default: "empty string"


scaleio-scheduler/config/config.go
package config

import (
	"flag"
	"strconv"

	xplatform "github.com/dvonthenen/goxplatform"
)

//consts exported out of package
const (
	//VersionInt in INT form
	VersionInt = 1

	//VersionStr in string form
	VersionStr = "0.3.0"

	//CPUPerMdmExecutor num of CPUs to MDM executor
	CPUPerMdmExecutor = 1.5

	//CPUPerNonExecutor num of CPUs to Non-MDM executor
	CPUPerNonExecutor = 0.5

	//MemPerMdmExecutor mem to MDM executor
	MemPerMdmExecutor = 3072

	//MemPerNonExecutor mem to Non-MDM executor
	MemPerNonExecutor = 512

	//DefaultRestPort rest port
	DefaultRestPort = 35000

	//RexrayRetry exponential backoff for retries
	RexrayRetry = 5

	//RexrayDelay a random delay to add to the exponential backoff
	RexrayDelay = 7
)

//consts internal
const (
	debMdm = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-mdm-2.0-10000.2072.Ubuntu.14.04.x86_64.deb"
	debSds = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-sds-2.0-10000.2072.Ubuntu.14.04.x86_64.deb"
	debSdc = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-sdc-2.0-10000.2072.Ubuntu.14.04.x86_64.deb"
	debLia = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-lia-2.0-10000.2072.Ubuntu.14.04.x86_64.deb"
	debGw  = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/emc-scaleio-gateway_2.0-10000.2072_amd64.deb"
	rpmMdm = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-mdm-2.0-10000.2072.el7.x86_64.rpm"
	rpmSds = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-sds-2.0-10000.2072.el7.x86_64.rpm"
	rpmSdc = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-sdc-2.0-10000.2072.el7.x86_64.rpm"
	rpmLia = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-lia-2.0-10000.2072.el7.x86_64.rpm"
	rpmGw  = "https://github.com/codedellemc/scaleio-framework/releases/download/v0.3.0/EMC-ScaleIO-gateway-2.0-10000.2072.x86_64.rpm"
	isoBin = "https://github.com/emccode/mesos-module-dvdi/releases/download/v0.4.6/libmesos_dvdi_isolator-1.0.1.so"
)

//Config is the representation of the config
type Config struct {
	LogLevel        string
	Debug           bool
	DeleteKeyValues bool
	DumpKeyValues   bool
	StoreAddKey     string
	StoreAddVal     string
	StoreDelKey     string
	Experimental    bool

	RexrayBranch  string
	RexrayVersion string

	IsolatorBinary string

	RestAddress          string
	RestPort             int
	MasterREST           string
	AltExecutorPath      string
	ExecutorMdmCPU       float64
	ExecutorNonCPU       float64
	ExecutorCPUFactor    float64
	ExecutorMdmMemory    float64
	ExecutorNonMemory    float64
	ExecutorMemoryFactor float64
	User                 string
	Hostname             string
	Role                 string
	Store                string
	StoreURI             string

	ClusterName          string
	ClusterID            string
	LbGateway            string
	ProtectionDomain     string
	StoragePool          string
	AdminPassword        string
	APIVersion           string
	PrimaryMdmAddress    string
	SecondaryMdmAddress  string
	TieBreakerMdmAddress string
	GatewayAddress       string
	DebMdm               string
	DebSds               string
	DebSdc               string
	DebLia               string
	DebGw                string
	RpmMdm               string
	RpmSds               string
	RpmSdc               string
	RpmLia               string
	RpmGw                string
}

//AddFlags adds flags to the command line parsing
func (cfg *Config) AddFlags(fs *flag.FlagSet) {
	fs.StringVar(&cfg.LogLevel, "loglevel", cfg.LogLevel,
		"Set the logging level for the application")
	fs.BoolVar(&cfg.Debug, "debug", cfg.Debug,
		"Debug mode prevents the reboot so the logs dont get cycled")
	fs.BoolVar(&cfg.DeleteKeyValues, "store.delete", cfg.DeleteKeyValues,
		"Helper function that deletes ScaleIO Framework Key/Value Store")
	fs.BoolVar(&cfg.DumpKeyValues, "store.dump", cfg.DumpKeyValues,
		"Helper function that dumps ScaleIO Framework Key/Value Store")
	fs.StringVar(&cfg.StoreAddKey, "store.add.key", cfg.StoreAddKey,
		"Modify a select store key")
	fs.StringVar(&cfg.StoreAddVal, "store.add.value", cfg.StoreAddVal,
		"Set the values to the key provided by store.key")
	fs.StringVar(&cfg.StoreDelKey, "store.del.key", cfg.StoreDelKey,
		"Delete a select store key")
	fs.BoolVar(&cfg.Experimental, "experimental", cfg.Experimental,
		"Sets the application to experimental mode")

	fs.StringVar(&cfg.RexrayBranch, "rexray.branch", cfg.RexrayBranch,
		"Which branch to grab the REX-Ray package from")
	fs.StringVar(&cfg.RexrayVersion, "rexray.version", cfg.RexrayVersion,
		"Which version to install from the provided branch")

	fs.StringVar(&cfg.IsolatorBinary, "isolator.binary", cfg.IsolatorBinary,
		"The URL for which Mesos Module DVDI to install")

	fs.StringVar(&cfg.RestAddress, "rest.address", cfg.RestAddress,
		"Mesos scheduler REST API address")
	fs.IntVar(&cfg.RestPort, "rest.port", cfg.RestPort, "Mesos scheduler REST API port")
	fs.StringVar(&cfg.MasterREST, "uri", cfg.MasterREST, "Mesos scheduler API URL")
	fs.StringVar(&cfg.AltExecutorPath, "executor.altpath", cfg.AltExecutorPath,
		"Provide an alternate path to the executor binary")
	fs.Float64Var(&cfg.ExecutorMdmCPU, "executor.cpu.mdm", cfg.ExecutorMdmCPU,
		"CPU resources to consume per-executor on MDM nodes")
	fs.Float64Var(&cfg.ExecutorNonCPU, "executor.cpu.non", cfg.ExecutorNonCPU,
		"CPU resources to consume per-executor on Non-MDM nodes")
	fs.Float64Var(&cfg.ExecutorCPUFactor, "executor.cpu.factor", cfg.ExecutorCPUFactor,
		"Fudge factor for effective CPU available. This allows overhead/reserve.")
	fs.Float64Var(&cfg.ExecutorMdmMemory, "executor.mem.mdm", cfg.ExecutorMdmMemory,
		"Memory resources (MB) to consume per-executor on MDM nodes")
	fs.Float64Var(&cfg.ExecutorNonMemory, "executor.mem.non", cfg.ExecutorNonMemory,
		"Memory resources (MB) to consume per-executor on Non-MDM nodes")
	fs.Float64Var(&cfg.ExecutorMemoryFactor, "executor.mem.factor", cfg.ExecutorMemoryFactor,
		"Fudge factor for effective memory available. This allows overhead/reserve.")
	fs.StringVar(&cfg.User, "user", cfg.User, "The User account the framework is running under")
	fs.StringVar(&cfg.Hostname, "hostname", cfg.Hostname, "The Hostname where the framework runs")
	fs.StringVar(&cfg.Role, "role", cfg.Role, "Framework role to register with the Mesos master")
	fs.StringVar(&cfg.Store, "store.type", cfg.Store, "The type of keyvalue store to use")
	fs.StringVar(&cfg.StoreURI, "store.uri", cfg.StoreURI, "Store URI to connect with")

	fs.StringVar(&cfg.ClusterName, "scaleio.clustername", cfg.ClusterName, "ScaleIO Cluster Name")
	fs.StringVar(&cfg.ClusterID, "scaleio.clusterid", cfg.ClusterID, "ScaleIO Cluster ID")
	fs.StringVar(&cfg.LbGateway, "scaleio.lbgateway", cfg.LbGateway, "Load Balanced IP/DNS Name")
	fs.StringVar(&cfg.ProtectionDomain, "scaleio.protectiondomain", cfg.ProtectionDomain,
		"ScaleIO Protection Domain Name")
	fs.StringVar(&cfg.StoragePool, "scaleio.storagepool", cfg.StoragePool,
		"ScaleIO StoragePool Name")
	fs.StringVar(&cfg.AdminPassword, "scaleio.password", cfg.AdminPassword,
		"ScaleIO Admin Password")
	fs.StringVar(&cfg.APIVersion, "scaleio.apiversion", cfg.APIVersion,
		"ScaleIO API Version")
	fs.StringVar(&cfg.PrimaryMdmAddress, "scaleio.preconfig.primary",
		cfg.PrimaryMdmAddress, "Pre-Configured Pri MDM Node. Requires Sec and TB "+
			"MDM nodes to be Pre-Configured")
	fs.StringVar(&cfg.SecondaryMdmAddress, "scaleio.preconfig.secondary",
		cfg.SecondaryMdmAddress, "Pre-Configured Sec MDM Node. Requires Pri and TB "+
			"MDM nodes to be Pre-Configured")
	fs.StringVar(&cfg.TieBreakerMdmAddress, "scaleio.preconfig.tiebreaker",
		cfg.TieBreakerMdmAddress, "Pre-Configured Tiebreaker MDM Node. Requires Pri "+
			"and Sec MDM nodes to be Pre-Configured")
	fs.StringVar(&cfg.GatewayAddress, "scaleio.preconfig.gateway",
		cfg.GatewayAddress, "Used to set a separated Gateway node. Otherwise, the "+
			"Primary MDM node is assumed to have the Gateway installed on it.")
	fs.StringVar(&cfg.DebMdm, "scaleio.ubuntu14.mdm", cfg.DebMdm, "ScaleIO MDM Package for Ubuntu 14.04")
	fs.StringVar(&cfg.DebSds, "scaleio.ubuntu14.sds", cfg.DebSds, "ScaleIO SDS Package for Ubuntu 14.04")
	fs.StringVar(&cfg.DebSdc, "scaleio.ubuntu14.sdc", cfg.DebSdc, "ScaleIO SDC Package for Ubuntu 14.04")
	fs.StringVar(&cfg.DebLia, "scaleio.ubuntu14.lia", cfg.DebLia, "ScaleIO LIA Package for Ubuntu 14.04")
	fs.StringVar(&cfg.DebGw, "scaleio.ubuntu14.gw", cfg.DebGw, "ScaleIO Gateway Package for Ubuntu 14.04")
	fs.StringVar(&cfg.RpmMdm, "scaleio.rhel7.mdm", cfg.RpmMdm, "ScaleIO MDM Package for RHEL7")
	fs.StringVar(&cfg.RpmSds, "scaleio.rhel7.sds", cfg.RpmSds, "ScaleIO SDS Package for RHEL7")
	fs.StringVar(&cfg.RpmSdc, "scaleio.rhel7.sdc", cfg.RpmSdc, "ScaleIO SDC Package for RHEL7")
	fs.StringVar(&cfg.RpmLia, "scaleio.rhel7.lia", cfg.RpmLia, "ScaleIO LIA Package for RHEL7")
	fs.StringVar(&cfg.RpmGw, "scaleio.rhel7.gw", cfg.RpmGw, "ScaleIO Gateway Package for RHEL7")
	fs.StringVar(&cfg.RpmMdm, "scaleio.centos7.mdm", cfg.RpmMdm, "ScaleIO MDM Package for CentOS7")
	fs.StringVar(&cfg.RpmSds, "scaleio.centos7.sds", cfg.RpmSds, "ScaleIO SDS Package for CentOS7")
	fs.StringVar(&cfg.RpmSdc, "scaleio.centos7.sdc", cfg.RpmSdc, "ScaleIO SDC Package for CentOS7")
	fs.StringVar(&cfg.RpmLia, "scaleio.centos7.lia", cfg.RpmLia, "ScaleIO LIA Package for CentOS7")
	fs.StringVar(&cfg.RpmGw, "scaleio.centos7.gw", cfg.RpmGw, "ScaleIO Gateway Package for CentOS7")
}

//NewConfig creates a new Config object
func NewConfig() *Config {
	ip, err := xplatform.GetInstance().Nw.AutoDiscoverIP()
	if err != nil {
		ip = "127.0.0.1"
	}

	return &Config{
		LogLevel:             env("LOG_LEVEL", "info"),
		Debug:                envBool("DEBUG", "false"),
		DeleteKeyValues:      envBool("DELETE_STORE", "false"),
		DumpKeyValues:        envBool("DUMP_STORE", "false"),
		StoreAddKey:          env("STORE_ADD_KEY", ""),
		StoreAddVal:          env("STORE_ADD_VAL", ""),
		StoreDelKey:          env("STORE_DEL_KEY", ""),
		Experimental:         envBool("EXPERIMENTAL", "false"),
		RexrayBranch:         env("REXRAY_BRANCH", "stable"),
		RexrayVersion:        env("REXRAY_VERSION", "latest"),
		IsolatorBinary:       env("ISOLATOR_BINARY", isoBin),
		RestAddress:          env("REST_ADDRESS", ip),
		RestPort:             envInt("REST_PORT", strconv.Itoa(DefaultRestPort)),
		MasterREST:           env("MESOS_MASTER_HTTP", "http://127.0.0.1:5050/api/v1/scheduler"),
		AltExecutorPath:      env("ALT_EXECUTOR_PATH", ""),
		ExecutorMdmCPU:       envFloat("EXECUTOR_MDM_CPU", strconv.FormatFloat(CPUPerMdmExecutor, 'f', -1, 64)),
		ExecutorNonCPU:       envFloat("EXECUTOR_NON_CPU", strconv.FormatFloat(CPUPerNonExecutor, 'f', -1, 64)),
		ExecutorCPUFactor:    envFloat("EXECUTOR_CPU_FACTOR", "1.0"),
		ExecutorMdmMemory:    envFloat("EXECUTOR_MDM_MEM", strconv.Itoa(MemPerMdmExecutor)),
		ExecutorNonMemory:    envFloat("EXECUTOR_NON_MEM", strconv.Itoa(MemPerNonExecutor)),
		ExecutorMemoryFactor: envFloat("EXECUTOR_MEMORY_FACTOR", "1.0"),
		User:                 env("USER", mesosUser()),
		Hostname:             env("HOSTNAME", mesosHostname()),
		Role:                 env("ROLE", "scaleio"),
		Store:                env("STORE_TYPE", "zk"),
		StoreURI:             env("STORE_URI", ""),
		ClusterName:          env("CLUSTER_NAME", "scaleio"),
		ClusterID:            env("CLUSTER_ID", ""),
		LbGateway:            env("LB_GATEWAY", ""),
		ProtectionDomain:     env("PROTECTION_DOMAIN", "default"),
		StoragePool:          env("STORAGE_POOL", "default"),
		AdminPassword:        env("ADMIN_PASSWORD", "Scaleio123"),
		APIVersion:           env("API_VERSION", "2.0"),
		PrimaryMdmAddress:    env("PRIMARY_MDM_ADDRESS", ""),
		SecondaryMdmAddress:  env("SECONDARY_MDM_ADDRESS", ""),
		TieBreakerMdmAddress: env("TIEBREAKER_MDM_ADDRESS", ""),
		GatewayAddress:       env("GATEWAY_ADDRESS", ""),
		DebMdm:               env("DEB_MDM", debMdm),
		DebSds:               env("DEB_SDS", debSds),
		DebSdc:               env("DEB_SDC", debSdc),
		DebLia:               env("DEB_LIA", debLia),
		DebGw:                env("DEB_GW", debGw),
		RpmMdm:               env("RPM_MDM", rpmMdm),
		RpmSds:               env("RPM_SDS", rpmSds),
		RpmSdc:               env("RPM_SDC", rpmSdc),
		RpmLia:               env("RPM_LIA", rpmLia),
		RpmGw:                env("RPM_GW", rpmGw),
	}
}
